YakimS/assignment3/WordSearcher.java

```diff
@@ -0,0 +1,104 @@
+package com.shaya;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class WordSearcher
+{
+
+    /**
+     * Given a grid of letters and a dictionary, find all the words from the
+     * dictionary that can be formed in the grid.
+     * The algorithm is case-sensitive ("Cat" != "CAT")
+     *
+     * The rules for forming a word:
+     * ● You can start at any position.
+     * ● You can move to one of the 8 adjacent cells (horizontally/vertically/diagonally).
+     * ● You can't visit the same cell twice in the same word.
+     *
+     *
+     * The algorithm iterate throw the grid cells to search for words starts with the
+     * char withing each cell.
+     *
+     * @param dictionary - a dictionary which contains the word and the prefixes of the words
+     * @param charsGrid -  grid of chars.
+     * @return set of all the words found in the grid that are also found in the dictionary.
+     */
+     public Set<String> allWordsInGrid(Dictionary dictionary, char[][] charsGrid)
+    {
+        HashSet<String> foundWords = new HashSet<>();
+        for(int x = 0 ; x < charsGrid.length ; x++)
+        {
+            for(int y = 0 ; y < charsGrid[0].length ; y++)
+            {
+                allWordsInGrid(dictionary,charsGrid ,x,y, "", foundWords);
+            }
+        }
+        return foundWords;
+    }
+
+    /**
+     *  This method will add prefix+char[x][y] to set of found words if it is in the dictionary.
+     *  Than search recursively all around the 8 adjacent cells to current location for another
+     *  char to add to the prefix+char[x][y], to create another candidate word.
+     *
+     *
+     * @param dictionary - a dictionary which contains the word and the prefixes of the words
+     * @param charsGrid -  grid of chars.
+     * @param x - marks the current X position in the grid
+     * @param y - marks the current Y position in the grid
+     * @param prefix - current prefix. The method will find words starting with this string.
+     * @param foundWords - set of all the words that are already found in the grid.
+     */
+    private void allWordsInGrid(Dictionary dictionary, char[][] charsGrid, int x, int y, String prefix, HashSet<String> foundWords)
+    {
+        if(     x < 0 || x >= charsGrid.length || y < 0 || y >= charsGrid[0].length  // if out of bound of the grid

```

Comment:
This is very odd formatting. Could you please use consistent formatting everywhere? It will make code a bit easier to read.